File: Web.ClientApp/src/shared/contracts/player-dto.ts
import {Card} from "./responses/added-card-response";

export interface IPlayerDto {
    id: string;
    roomId: string;
    isLeader: boolean;
    readiness: boolean;
    playerName: string;
    score: number;
    avatarUrl: string;
    cards: Array<Card>;
    move: boolean;
    money: number;
    inGame: boolean;
    online: boolean;
}

export type Player = IPlayerDto;

File: src/Services/App/App.Contracts/Data/PlayerDto.cs
using App.Domain.Entities;

namespace App.Contracts.Data;

public sealed record PlayerDto(
    Guid Id,
    Guid RoomId,
    bool IsLeader,
    bool Readiness,
    string PlayerName,
    int Score,
    string AvatarUrl,
    IEnumerable<Card> Cards,
    bool Move,
    int Money,
    bool InGame,
    bool Online);